### ISSUE TITLE
[a11y Support] Benefits: Conditions Team - Autocomplete (Office Hours Support)

### ISSUE THREAD

**Issue by sara-amanda** [department-of-veterans-affairs/va.gov-team#96763]
> [!NOTE]
> **QUICK SUMMARY:** `
> - **Conditions Team Support:** ~Combo Box~ Autocomplete
> - **Office Hours Support Regarding**: ~Combo box,~ **Autocomplete** values announced and available results.
> - **Original Support Source**: [Office Hours](https://oddball.slack.com/archives/C05LB36ATAQ/p1729024504107479?thread_ts=1729011011.019479&cid=C05LB36ATAQ) c/o @mutiaadhocteam 
> - **Updates**: 
>     - This ticket body contains dated updates in the _Issue Description_ section.  
>     - The UX has been updated, along with accessibility improvements. 
>     - The team is planning research for December ref. ticket #96633 

### Original Office Hours Request

**Date**: 10/15/2024

<details><summary>Review the original office hours request</summary>

#### Office Hours Submission 10/15/2024

> - **Name**: William Phelps
> - **Team**: Conditions Team
> - **Details**: Refactored src/applications/disability-benefits/all-claims/components/ComboBox.jsx and with that have aimed to improve aspects of its implementation including:
>     - Name: Updated to “AutoComplete” since was told our team had been informed that ComboBox was not the correct name. Looking for advice for the best name for this component.
>     - Debounce: implemented 200ms debounce (previously it was intended to be implemented but was not being implemented)
>     - Highlights freeText result when typing
>     - Keeps focus in the input as arrow through results as suggested here [William Phelps: "Hello! Our team has a question about what ..."](https://dsva.slack.com/archives/C8E985R32/p1728398775821319) (previously focus moved into the list)
>     - Fixes issues that when user blurred input, input was automatically refocused
>     - Fixes issue that when user adds value then blurs input then re-focuses input, the list would not be shown until the user changed the value.
>     - Fixes test case of "If click ‘Add another condition’ while another condition is in edit mode, focus should jump to the new input field"
>     - Fixes arrow down in list resulted in a jolting and confusing experience.
>     - Removed aria-atomic from our aria-live
> - **Agenda**
>     - Name: ComboBox vs AutoComplete
>     - Aria-live being interrupted
>     - Aria-live content - should it mention the first item
> - **Files**:
>     - [2024-03 IAC test cases](https://docs.google.com/spreadsheets/d/12DaRQF6iwm_rUOhmgMht0h00HTJPoIa0YMxVCmHhzLU/edit?gid=1789687425#gid=1789687425)
> - **Links**:
>     - PR - https://github.com/department-of-veterans-affairs/vets-website/pull/32311

</details>

### Feedback from Maria Post-Office Hours

**Date**: 10/15/2024

<details><summary>Review the feedback from office hours</summary>

[Feedback from Maria](https://oddball.slack.com/archives/C05LB36ATAQ/p1729024504107479?thread_ts=1729011011.019479&cid=C05LB36ATAQ) Post Office Hours

>     - Support needed for the Conditions team!
>     - A ticket will be made soon in regards to the naming of a combo box as well as an odd behavior that occurs with the announcing of the value present in the combo box and its available results.
>     - After a ticket is made and sent to us, he and I will likely work together to ensure that the software behavior is not nightmarish via CodeSpace! As long as he bypasses log in on his end, all should be well from testing end.

</details>

### AutoComplete Components and Screen Readers

**Date**: 10/24/2024

<details><summary>Review the Slack "Accessibility-Help" channel thread</summary>

Refer to the [Autocomplete Components and Screen Readers Thread Link in DSVA Slack](https://dsva.slack.com/archives/C8E985R32/p1729808871239759). The convo is also pasted below for quick reference. 

#### William Phelps: 
> **Quick question about screen readers and AutoComplete components:**  Both these AutoComplete examples do not allow focusing the screen reader  using control+option+rightArrow on the listbox and options instead the user uses up and down arrows to interact and hear about the options. Does this inability to use control+option+rightArrow then control+option+downArrow to go into the listbox group detract from its accessibility?
> - https://alphagov.github.io/accessible-autocomplete/examples/
> - https://www.downshift-js.com/downshift (the demo components are underneath the code blocks)

#### Jamie Klenetsky Fay
> i’m pretty sure that that’s expected behavior - up + down to navigate through the list, right + left to move the text cursor within the input field. check out the keyboard support section on [W3C’s page about comboboxes](https://www.w3.org/WAI/ARIA/apg/patterns/combobox/)

#### Homer Gaines
> Correct. This is expected behavior.

#### William Phelps
> Thank you! Slightly tangential but what would be a good name for a component in vets-website that has the functionality of the first example within https://alphagov.github.io/accessible-autocomplete/examples/? Previously we called the component ComboBox but I heard the note that in staging review there was helpful feedback (potentially from 
> [@Matthew Dingee](https://dsva.slack.com/team/U03740NA3E3) that there was a better name for a component that does the following:
> - Supports text input (this makes it "editable" in the above document)
>> - "When the popup is triggered, it presents suggested values. If the combobox is editable, the suggested values complete or logically correspond to the characters typed in the combobox. The character string the user has typed will become the value of the combobox unless the user selects a value in the popup." ("List autocomplete with manual selection" in the above document)
> - Would it be Autocomplete?

#### Matthew Dingee
> Yes, to me that first example is an autocomplete. Combo box is a combination of a text-input and a select. Thus if the field doesn't allow me to open a menu, scroll a menu of options, and make a selection then it does not have select properties and thus can't be a Combo box. There are some Combo boxes further down that list which have an affordance for being able to open a menu of options and make a selection.

#### William Phelps
> Appreciate that - that makes sense!

</details>

### Updates to Autocomplete from Conditions Team

**Date**: 11/1/2024

<details><summary>Review the update on autocomplete</summary>

[c/o Maria in Slack](https://oddball.slack.com/archives/C05LB36ATAQ/p1730468284491139)

> Hi all! Just received this from William Phelps from the Conditions Team in regards to their combobox ...
> Thank you so much for following up. Today, I merged the All Claims Refactor Combo Box pull request and so now the updated UI is accessible on staging (but because there is a feature toggle it is not yet accessible on production to real users). The updated component, now called Autocomplete, fixed the following from the previous version:
> - Adds a 200ms delay before the user gets results which aims to create a smoother user experience so that results don't immediately appear after each character press. It also reduces unnecessary re-renders of the component.
> - Highlights freeText result when user types (the first result of our Autocomplete is what the user has typed in so far and we call this the freeText result)
> - Removes bolding of text within freeText result
> - Keeps focus in the input as the users arrows through results
> - Fixes issues that previously when user blurred the input the input was automatically refocused
> - Fixes issue that when user adds value then blurs input then re-focuses input, the list would not be shown until the user changed the value. Now it re-shows results on re-focus.
> - It also aimed to simplify and refine the screenreader experience. Our accessibility engineer Tracy Tran tested it but I was excited to have your expertise. Let me know about next steps or how I can assist in directing you to the Autocomplete.

[Related 10/24/2024 DSVA Slack Thread](https://dsva.slack.com/archives/C8E985R32/p1729808871239759)

</details>

## About the Team

- **Team name**:  Conditions Team
- **Support Item:**  Autocomplete
- **Product manager:**  Yurena Garcia  
- **FE engineer:** William Phelps  (attended office hours)
- **Slack Channel:** [#benefits-contention-classification](https://dsva.slack.com/archives/C04AZ8T7XN1  )

```[tasklist]
### a11y Tasks
- [x] Collaborated with the Conditions Team during Office Hours (pursuant to request to meet) 
- [x] a11y Support provided via Office hours on 10/15/2024 and follow-ups were made in Slack. 
- [x] Documented in this ticket
- [x] Responded and requested related tickets
```
```[tasklist]
### Acceptance Criteria
- [x] Name change from Combo Box to Autocomplete to avoid confusion with combo box component being built at this time. (Confirmed) 11/1/2024
- [x] UX Updates 11/1/2024
- [x] Accessibility Updates 11/1/2024
```
```[tasklist]
## Ticket Updates (CAIA Internal)
- [ ] _Connect to an `Epic` or `Intake` (what body of work is this a part of?)_
- [ ] _Label with `Originator/Team` (product team or stakeholder requesting support)_
- [ ] _Label date in the `Open Date` field_
- [ ] _Label with `Estimate` (level of effort expected for this ticket)_
- [ ] _Add `Assignee(s)` name(s) to ticket_ (include Naomi)
- [ ] _Add `Assignee(s) name(s) to each task` they will complete via handle tag (if known)_
- [ ] _Select a `Priority Level`_
- [ ] _Update date in `Last Checked` field_
- [ ] _Label with `Actual` (level of effort it took to complete this ticket)_
- [ ] _Update date in_ `Closed Date`
```

```[tasklist]
### Related Items
- [ ] #32311 Conditions 599 Refactor ComboBox](https://github.com/department-of-veterans-affairs/vets-website/pull/32311) Merged Prerequisite - Merge this vets-api PR to create the feature toggle - [Conditions 599 Add "Add Disabilities Enhancement" feature toggle #18983](https://github.com/department-of-veterans-affairs/vets-api/pull/18983)
- [ ] #96633
```
